blueprints/movies.py

```diff
@@ -29,40 +36,133 @@ def parse_movie(data):
     return movie
 
 
+def init_cache():
+    global cache_limit, mcache
+    i = 1
+    while i <= cache_limit:
+        mcache[i] = {}
+        mcache[i]['id'] = -1
+        i += 1
+
+
 @movies.route('/<int:movie_id>', methods=['GET'])
 def get_movie(movie_id):
-    movie = current_app.movies.get_movie(movie_id)
+    global lock, cinit
+    if not cinit:
```

Comment:
Ezt ugyanúgy lockkal kell védeni, különben többször inicializálhat, akár már meglévő cachet is felülírhat. Itt érdemes lett volna double check lock patternt használni.